R/functions.R

```diff
@@ -22,7 +22,7 @@ form_calculation_limits <- function(data, counter, periodMin, chartType = "C", m
   
   exclusion_points <- find_extremes(data, chartType, counter, periodMin, maxNoOfExclusions, rule2Tolerance)
   
-  calculation_period <- data[counter:(counter + periodMin),]
```

Comment:
@ImogenConnorHelleur this was the main problem I spotted. As it was, suppose periodMin = 21 and counter = 22. Then this calculation period is `data[22:43]`, which is 43 - 22 + 1 = 22 rows long, not 21. Hence the amendment to subtract one from the end row number.

With counter = 0 (which really means our starting row for calculation is row 1, not "row 0"), the original code gives calculation period `data[0:21]` which is equivalent to `data[1:21]`, the correct result (note e.g. that for a data frame `df`, `df[0:1,]` gives the same result as `df[1:1,]`).

Of course, fixing this line so that it works for non-zero values of `counter` meant changing the starting value of the counter from 0 to 1, which in turn meant changing a whole bunch of other lines of code to work with the new counter "system". That accounts for the majority of the diff.

Reply:
Hi @HorridTom , this makes sense to me now! And it solves the mysterious problem that I'd been having with an extra row appearing at the end of some output limits tables and not others. I've checked through the code and this seems to have fixed the problem.